### ISSUE TITLE
Change in the Dimension (shape) because of np.hstack on tf.keras.preprocessing.text.Tokenizer.texts_to_sequences

### ISSUE THREAD

**Issue by rmothukuru** [tensorflow/tensorflow#36779]
**System information** 

- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes

- OS Platform and Distribution (e.g.,Linux Ubuntu 16.04): - NA, as it can be replicated in Google Colab, irrespective of an OS

Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: - NA

TensorFlow installed from (source or binary): - pip install tensorflow==2.1

TensorFlow version (use command below): - 2.1

Python version: - Google Colab Python 3

Bazel version (if compiling from source): NA

GCC/Compiler version (if compiling from source): - CUDA/cuDNN version: NA

- GPU model and memory: NA

**Describe the current behavior**: TLDR, working properly for Testing (Validation) Labels but shape is changing for Training Labels.

I have applied np.hstack on `tensorflow.keras.preprocessing.text.Tokenizer.texts_to_sequences` for both Training Labels and for Validation (Testing) Labels.

Surprisingly and mystically, the Size of the Output, after I applied on Training Labels is different from that of before I have applied `np.hstack`. However, there is no Change in the Shape for Validation Labels, before and after the application of `tensorflow.keras.preprocessing.text.Tokenizer.texts_to_sequences` and `np.hstack`.

**Describe the expected behavior**: There should not be change in the Shape before and after application of above Methods and the behavior should be same for both Training Labels and Testing Labels.

**Code to reproduce the issue** :  This is the Link of the [Github Gist](https://colab.sandbox.google.com/gist/rmothukuru/fa586147ebefdc9f2eb9ff03206e5787/discrepancy_keras_tokenizer.ipynb), to reproduce the error easily.

Complete Code to reproduce the Error is given below (just in case if the link doesn't work):

```
!pip install tensorflow==2.1

# For Preprocessing the Text => To Tokenize the Text
from tensorflow.keras.preprocessing.text import Tokenizer
# If the Two Articles are of different length, pad_sequences will make the length equal
from tensorflow.keras.preprocessing.sequence import pad_sequences

# Package for performing Numerical Operations
import numpy as np

Unique_Labels_List = ['India', 'USA', 'Australia', 'Germany', 'Bhutan', 'Nepal', 'New Zealand', 'Israel', 'Canada', 'France', 'Ireland', 'Poland', 'Egypt', 'Greece', 'China', 'Spain', 'Mexico']


Train_Labels = Unique_Labels_List[0:14]
#print('Train Labels = {}'.format(Train_Labels))

Val_Labels =  Unique_Labels_List[14:]
#print('Val_Labels = {}'.format(Val_Labels))

No_Of_Train_Items = [248, 200, 200, 218, 248, 248, 249, 247, 220, 200, 200, 211, 224, 209]
No_Val_Items = [212, 200, 219]

T_L = []
for Each_Label, Item in zip(Train_Labels, No_Of_Train_Items):
    T_L.append([Each_Label] * Item)

T_L = [item for sublist in T_L for item in sublist]

V_L = []
for Each_Label, Item in zip(Val_Labels, No_Val_Items):
    V_L.append([Each_Label] * Item)

V_L = [item for sublist in V_L for item in sublist]


len(T_L)

len(V_L)

label_tokenizer = Tokenizer()

label_tokenizer.fit_on_texts(Unique_Labels_List)

# Since it should be a Numpy Array, we should Convert the Sequences to Numpy Array, for both Training and 
# Test Labels

training_label_list = label_tokenizer.texts_to_sequences(T_L)

validation_label_list = label_tokenizer.texts_to_sequences(V_L)

training_label_seq = np.hstack(training_label_list)

validation_label_seq = np.hstack(validation_label_list)

print('Actual Number of Train Labels before np.hstack are {}'.format(len(training_label_list)))
print('Change in the Number of Train Labels because of np.hstack are {}'.format(len(training_label_seq)))

print('-------------------------------------------------------------------------------------------------------')

print('Actual Number of Validation Labels before np.hstack are {}'.format(len(validation_label_list)))
print('However, there is no change in the Number of Validation Labels because of np.hstack {}'.format(len(validation_label_seq)))
```

**Other info / logs** :

**Comment by gadagashwini** [comment]
@rmothukuru, Can you refer similar issue resolution on stackoverflow https://stackoverflow.com/questions/60237754/change-in-the-dimension-shape-because-of-np-hstack-on-tf-keras-preprocessing-t. Thanks!
